zerver/views/presence.py

```diff
@@ -57,6 +57,9 @@ def update_active_status_backend(request, user_profile, status=REQ(),
                                  ping_only=REQ(validator=check_bool, default=False),
                                  new_user_input=REQ(validator=check_bool, default=False)):
     # type: (HttpRequest, UserProfile, str, bool, bool) -> HttpResponse
+    if user_profile.is_bot:
+        return json_error(_('Presence cannot be updated for bot users.'))
```

Comment:
non-unique is best unless there's a good reason to distinguish the cases.  It saves an extra string to translate.